android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2608,6 +2608,206 @@ public static int getKeyboardHeightMin(Context context, int orientation) {
+   * Uses a live (unsaved) height for the specified orientation and reads saved height for the other.
+   * This is useful for displaying real-time percentages during keyboard height adjustment.
+   *
+   * <p>The percentage is calculated using Math.ceil() to ensure consistency with
+   * {@link #createKeyboardHeightString(Context, String, String)} and
+   * {@link #getKeyboardHeightPercentage(Context, int)}.
+   *
+   * <p>Example usage during drag operation:
+   * <pre>{@code
+   * // User is dragging to adjust portrait height
+   * int currentOrientation = KMManager.getOrientation(context);
+   * String display = KMManager.createKeyboardHeightString(
+   *     context,
+   *     currentHeight,  // Live height being adjusted
+   *     currentOrientation,
+   *     "Portrait",
+   *     "Landscape"
+   * );
+   * // Result: "145% Portrait | 100% Landscape"
+   * }</pre>
+   *
+   * @param context Context
+   * @param liveHeight The current keyboard height in pixels (not yet saved to SharedPreferences)
+   * @param liveOrientation The orientation for which liveHeight applies
+   *                        (Configuration.ORIENTATION_PORTRAIT or Configuration.ORIENTATION_LANDSCAPE)
+   * @param portraitLabel Label for portrait orientation (e.g., "Portrait")
+   * @param landscapeLabel Label for landscape orientation (e.g., "Landscape")
+   * @return String in format "100% Portrait | 100% Landscape"
+   * @see #createKeyboardHeightString(Context, String, String)
+   * @see #getKeyboardHeightPercentage(Context, int)
+   * @see #calculateKeyboardHeightFromTouch(Context, int, int)
+   */
+  public static String createKeyboardHeightString(Context context, int liveHeight, int liveOrientation,
+                                                   String portraitLabel, String landscapeLabel) {
+    int portraitPercent = 100;
+    int landscapePercent = 100;
+
+    // Calculate percentage for the live orientation
+    int liveDefaultHeight = getDefaultKeyboardHeight(liveOrientation);
+    if (liveDefaultHeight > 0 && liveHeight > 0) {
+      int livePercent = (int) Math.ceil((liveHeight * 100.0) / liveDefaultHeight);
+      if (livePercent == 0) {
+        livePercent = 100;
+      }
+
+      if (liveOrientation == Configuration.ORIENTATION_PORTRAIT) {
+        portraitPercent = livePercent;
+        landscapePercent = getKeyboardHeightPercentage(context, Configuration.ORIENTATION_LANDSCAPE);
+      } else {
+        landscapePercent = livePercent;
+        portraitPercent = getKeyboardHeightPercentage(context, Configuration.ORIENTATION_PORTRAIT);
+      }
+    } else {
+      // Fallback: read both from saved preferences
+      portraitPercent = getKeyboardHeightPercentage(context, Configuration.ORIENTATION_PORTRAIT);
+      landscapePercent = getKeyboardHeightPercentage(context, Configuration.ORIENTATION_LANDSCAPE);
+    }
+
+    return portraitPercent + "% " + portraitLabel + " | " +
+           landscapePercent + "% " + landscapeLabel;
```

Comment:
Some languages may want different orders (e.g. we might see "portrait %100 | landscape %120"); this really should be a format string for localization

Reply:
We shouldn't be constructing this same string in two places. We need to DRY it out.